pageserver/src/tenant/tasks.rs

```diff
@@ -346,6 +346,7 @@ async fn gc_loop(tenant: Arc<Tenant>, cancel: CancellationToken) {
             RequestContext::todo_child(TaskKind::GarbageCollector, DownloadBehavior::Download);
 
         let mut first = true;
+        tenant.gc_block.set_lsn_lease_deadline(tenant.get_lsn_lease_length());
```

Comment:
Ahaa, now I see, resolving my earlier thread.

I think I can now see why you needed to add so many of those `lsn_lease_length: 0` or the opposite.

Earlier we slept strictly for `lsn_lease_length` and random_init_delay. With this deadline being captured here, we are sleeping for `max(lsn_lease_length, random_init_delay)` which does seem right but is different.

I guess I need to try this... I am not inclined to make a startup changing timing change in this PR, but I still don't have a clue why you needed to add so many of those test changes.

Reply:
Aah oki now I understand the test changes: because the gc blocking now handles the "sleep", it affects manual calls as well.

Reply:
Only thing I am now wondering is that will this be a footgun:

Someone uses PageserverHttpClient.timeline_gc and nothing happens in their test. The pageserver log will have the skipping reason, but I wonder if it's enough.

Are there any tests in which we do gc and expect it to not be skipped? We could make this less of a footgun/timesink by describing this decision in the return value, and asserting it on the python side (with some default `allow_skipped=False` parameter?).

Reply:
> Earlier we slept strictly for `lsn_lease_length` and random_init_delay. With this deadline being captured here, we are sleeping for `max(lsn_lease_length, random_init_delay)` which does seem right but is different.

I think right now it only sleeps for `random_init_delay`. Do we want to make it `max(lsn_lease_length, random_init_delay)`? I think it will make a difference. But yes, it also affects manual gc.

Reply:
> Are there any tests in which we do gc and expect it to not be skipped? We could make this less of a footgun/timesink by describing this decision in the return value, and asserting it on the python side (with some default `allow_skipped=False` parameter?).

Yes, some of them do this. If we are skipping GC, GcResult::layers_total will be 0. I will use this on Python side for asserting.

Reply:
Created a followup: https://github.com/neondatabase/neon/issues/9063